crates/scheduler/src/bin/hypha-scheduler.rs

```diff
@@ -231,12 +232,30 @@ async fn run(config: ConfigWithMetadata<Config>) -> Result<()> {
             .collect::<Vec<_>>();
         let parameter_server = &mut allocated_parameter_servers[0];
 
+        // Find the data provider for the requested dataset
+        let tensor_name = "data_6";
+        let data_provider_peer_id = *network
+            .find_provider(tensor_name)
+            .await
+            .expect("data providers")
+            .into_iter()
+            .collect::<Vec<_>>()
+            .first()
+            .expect("a data provider");
+
         for worker in &mut allocated_workers {
             worker
                 .dispatch(JobSpec {
                     executor: Executor::DiLoCoTransformer {
-                        model: Fetch::uri("EleutherAI/gpt-neo-125m"), // Fetch::uri("https://example.com/"),
-                        data: Fetch::uri("datablations/c4-filter-small"),
+                        model: Fetch::uri("EleutherAI/gpt-neo-125m"),
+                        data: Fetch::data_peer(
+                            data_provider_peer_id,
+                            DataSlice {
+                                dataset: tensor_name.to_string(),
+                                // TODO: determine slice offsets for data-parallel training with the available workers
+                                offsets: (0, 6),
```

Comment:
To slice a dataset, we first need to know its size. I.e., we need either an API in `hypha-data` to retrieve that size or include it as part of what's stored in the DHT.